time.js

```diff
@@ -8,21 +8,321 @@
+        }
+
+        // Duration string
+        catch(e) {
+            try {
+                return time.ZonedDateTime.now().plus(time.Duration.parse(str));
+            }
+
+            // Unsupported
+            catch(e) {
+                throw Error('"' + str + '" cannot be parsed into a format that can be converted to a ZonedDateTime');
+            }
+        }
+    }
+
+}
+
+/**
+ * Converts the state of the passed in Item to a time.ZonedDateTime
+ * @param {items.Item} item 
+ * @returns {time.ZonedDateTime}
+ * @throws error if the Item's state is not supported or the Item itself is not supported
+ */
+const convertItem = function(item) {
+
+
+    // Uninitialized
+    if(item.isUninitialized) {
+        throw Error('Item ' + item.name + ' is NULL or UNDEF, cannot convert to a time.ZonedDateTime');
+    }
+
+    // Number type Items
+    else if(item.rawState instanceof DecimalType) {
+        return addMillisToNow(item.rawState.floatValue());
+    }
+
+    // String type Items
+    else if(item.rawState instanceof StringType) {
+        return parseString(item.state);
+    }
+
+    // DateTime Items
+    else if(item.rawState instanceof DateTimeType) {
+        return javaZDTtoZDT(item.rawState.getZonedDateTime());
+    }
+
+    // Number:Time type Items
+    else if(item.rawState instanceof QuantityType) {
+        return addQuantityType(item.rawState);
+    }
+
+    else {
+        throw Error(item.type + ' is not supported for conversion to time.ZonedDateTime');
+    }
+}
+
+/**
+ * Converts the passed in when to a time.ZonedDateTime based on the following 
+ * set of rules.
+ * 
+ * - null, undefined: time.ZonedDateTime.now()
+ * - time.ZonedDateTime: unmodified
+ * - Java ZonedDateTime, DateTimeType: converted to time.ZonedDateTime equivalnet
+ * - JavaScript native Date: converted to a time.ZonedDateTime using SYSTEM as the timezone
+ * - number, bigint, Java Number, DecimalType: rounded and added to time.ZonedDateTime.now() as milliseconds
+ * - QuantityType: if the units are Time, added to now
+ * - Item: converts the state of the Item based on the *Type rules described here
+ * - String, Java String, StringType: Parsed based on the following rules
+ *     - ISO 8601: Not yet implemented
+ *     - RFC (output from a Java ZonedDateTime.toString()): parsed to time.ZonedDateTime
+ *     - HH:mm[:ss] (i.e. 24 hour time): that time with today's date (seconds are optional)
+ *     - KK:mm[:ss][ ][aa] (i.e. 12 hour time): that time with today's date (seconds and space between time and am/pm are optional)
+ *     - Duration: any duration string supported by time.Duration (e.g. 'PT5H4M3.210S'), see https://js-joda.github.io/js-joda/class/packages/core/src/Duration.js~Duration.html#static-method-parse
+ * @param {*} when any of the types discussed above
+ * @returns time.ZonedDateTime
+ * @throws error if the type, format, or contents of when are not supported
+ */
+time.toZDT = function(when) {
```

Comment:
This would be a really easy and awesome method to have some tests around, so pass in each of the listed dates in the comment and verify you get the right zdt time back.

Reply:
lol, i just re-read your original post and saw you had a lot of testing!   Can you add those tests to this library ?  There is a test directory with some tests around the rule builder, but should provide a good template for adding the testing logic you have already written for the time classes

Reply:
Still trying to figure out how to make Mocha work. I've installed the libraries as global but they are not showing up. In short I've not even managed to run the tests that are already there. I'm still muddling through but I've held this PR up long enough already. Maybe it's worth coming back later with a new PR.